nixos/modules/hardware/cpu/intel-msr.nix

```diff
@@ -0,0 +1,58 @@
+{ lib
+, config
+, options
+, ...
+}:
+let
+  inherit (builtins) hasAttr;
+  inherit (lib) mkIf mdDoc;
+  cfg = config.hardware.cpu.intel.msr;
+  opt = options.hardware.cpu.intel.msr;
+  defaultGroup = "msr";
+  isDefaultGroup = cfg.group == defaultGroup;
+  set = "to set for devices of the `msr` kernel subsystem.";
+in
+{
+  options.hardware.cpu.intel.msr = with lib.options; {
```

Comment:
why is it intel specific?

Reply:
Hmm... I only came across references to Intel CPUs, but you are right, there seem to be AMD CPUs that support it. Which name do you suggest? `hardware.cpu.x86.msr`?

Reply:
Renamed to `hardware.cpu.x86.msr`.